GEOSldas_App/setup_utils.py

```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+from collections    import OrderedDict
+
+def parseInputFile(inpfile, ladas_cpl=0):
+   """ 
+     Parse the input file and return a dict of options
+     Inpfile : input file
+     Output  : dict, if inpfile does not exist, return empty {}
+   """ 
+   if not os.path.exists(inpfile):
+     assert ladas_cpl > 0, " No exeinput file only if ladas_cpl > 0"
+     return {}
+        
+   inpdict = OrderedDict()
+   errstr  = "line [%d] of [%s] is not in the form 'key: value'"
+   # determine which default values to pick from GEOS_SurfaceGridComp.rc
+   # must be in the format '# GEOSldas=>' or # GEOSagcm=>
+   if ladas_cpl == 0 :
+      use_rc_defaults = '# GEOSldas=>'    # use defaults for LDAS
+   else :
+      use_rc_defaults = '# GEOSagcm=>'    # use defaults for AGCM
+            
+   linenum = 0
+   with open(inpfile) as fin:
+     for line in fin:
+       line = line.strip()
+       linenum += 1
+       if not line:
+          continue
+       if line.startswith(use_rc_defaults):
+          line = line.split(use_rc_defaults)[1]
+       # handle comments
+       position = line.find('#')
+       if position==0: # comment line
+          continue
+       if position>0:  # strip out comment
+          line = line[:position]
+       # we expect a line to be of the form
+       # key : value
+       assert ':' in line, errstr % (linenum, inpfile)
+                                                               
+       key, val = line.split(':',1)
+       key = key.strip()
+       val = val.strip()
+       if not key or not val:
+         print ("WARNING: " + errstr % (linenum, inpfile))
+         continue
+      #raise Exception(errstr % (linenum, inpfile))
+       if key in inpdict:
+         raise Exception('Duplicate key [%s] in [%s]' % (key, inpfile))
+       inpdict[key] = val.strip()
+   return inpdict
+
+def echoInputFile(inpfile, ladas_cpl = 0):
+   """
+    Echo inpfile, ignore line starts with "## "
+   """
+   if ladas_cpl == 0 :
+      use_rc_defaults = '# GEOSldas=>'    # use defaults for LDAS
+   else :
+      use_rc_defaults = '# GEOSagcm=>'    # use defaults for AGCM
+
+   with open (inpfile) as fin :
+     for line in fin:
+        if line.startswith("## "):
```

Comment:
Similar to the comment above, it's risky to identify the line using the '## ' string.  Perhaps we simply keep these lines in the output file?  They don't hurt there.